### ISSUE TITLE
Error "out of memory" while there is enough memory to allocate

### ISSUE THREAD

**Issue by Jonglastra** [deforum-art/deforum-stable-diffusion#180]

**Comment by Jonglastra** [comment]
Hi, all. 2-3 weeks ago I was able to make videos with deforum-auto1111 in 1280 x 768 (I know that it is not the normal basis for the models but it is ok in my case, it was space video). Recently I made the update of deforum in the extension tab, and since the program doesn't succeed to be stable, I can generate images (in 1280 x 768) and even parts of videos (before it crashes) but systematically it bugs at some point. Is it because it has been adapted to torch 2.0.0 ? I cannot for the moment use it (I have a 5700xt  and I have issues with the new torch version, https://github.com/AUTOMATIC1111/stable-diffusion-webui/issues)/8139). 


################################################################
Install script for stable-diffusion + Web UI
Tested on Debian 11 (Bullseye)
################################################################

################################################################
Running on stef user
################################################################

################################################################
Repo already cloned, using it as install directory
################################################################

################################################################
Create and activate python venv
################################################################

################################################################
Launching launch.py...
################################################################
Python 3.10.6 (main, Mar 10 2023, 10:55:28) [GCC 11.3.0]
Commit hash: 22bcc7be428c94e9408f589966c2040187245d81
Installing requirements for Web UI

Launching Web UI with arguments: --no-half --medvram --always-batch-cond-uncond
No module 'xformers'. Proceeding without it.
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
[AddNet] Updating model hashes...
0it [00:00, ?it/s]
Loading weights [6ce0161689] from /home/stef/stable-diffusion-webui/models/Stable-diffusion/v1-5-pruned-emaonly.safetensors
Creating model from config: /home/stef/stable-diffusion-webui/configs/v1-inference.yaml
LatentDiffusion: Running in eps-prediction mode
DiffusionWrapper has 859.52 M params.
Applying cross attention optimization (Doggettx).
Textual inversion embeddings loaded(3): EmWat69, alexandradaddario, anyachalotra
Model loaded in 155.2s (load weights from disk: 3.7s, create model: 1.5s, apply weights to model: 134.5s, apply dtype to VAE: 0.9s, load VAE: 10.9s, move model to device: 0.1s, hijack: 0.9s, load textual inversion embeddings: 2.7s).
Running on local URL:  http://127.0.0.1:7861

To create a public link, set `share=True` in `launch()`.
Startup time: 197.2s (import torch: 5.7s, import gradio: 6.5s, import ldm: 1.3s, other imports: 3.0s, list SD models: 0.1s, setup codeformer: 0.2s, load scripts: 0.5s, load SD checkpoint: 155.4s, opts onchange: 0.2s, reload hypernets: 0.2s, create ui: 7.2s, gradio launch: 16.8s).
Deforum extension for auto1111 webui, v2.3b
Git commit: 82f39f9c (Fri Apr 21 19:33:10 2023)
Saving animation frames to:
/home/stef/stable-diffusion-webui/outputs/img2img-images/Deforum_20230422102033
Animation frame: 0/40  
Seed: 3826518503
Prompt: tiny cute swamp bunny, highly detailed, intricate, ultra hd, sharp photo, crepuscular rays, in focus, by tomasz alen kopera
Not using an init image (doing pure txt2img)
╭─────┬───┬───────┬─────┬──────┬────┬────╮
│Steps│CFG│Denoise│Angle│ Zoom │Tr X│Tr Y│
├─────┼───┼───────┼─────┼──────┼────┼────┤
│ 25  │7.0│ None  │  0  │1.0025│ 0  │ 0  │
╰─────┴───┴───────┴─────┴──────┴────┴────╯
  0%|                                                    | 0/25 [00:00<?, ?it/s]MIOpen(HIP): Warning [SQLiteBase] Missing system database file: gfx1030_18.kdb Performance may degrade. Please follow instructions to install: https://github.com/ROCmSoftwarePlatform/MIOpen#installing-miopen-kernels-package
100%|███████████████████████████████████████████| 25/25 [02:14<00:00,  5.38s/it]
Animation frame: 2/40  ███                     | 25/196 [01:11<08:13,  2.89s/it]
Creating in-between cadence frame: 0; tween:0.50;
Creating in-between cadence frame: 1; tween:1.00;
Seed: 3826518504
Prompt: tiny cute swamp bunny, highly detailed, intricate, ultra hd, sharp photo, crepuscular rays, in focus, by tomasz alen kopera
╭─────┬───┬───────┬─────┬─────┬────┬────╮
│Steps│CFG│Denoise│Angle│Zoom │Tr X│Tr Y│
├─────┼───┼───────┼─────┼─────┼────┼────┤
│ 25  │7.0│ 0.35  │  0  │1.003│ 0  │ 0  │
╰─────┴───┴───────┴─────┴─────┴────┴────╯
100%|█████████████████████████████████████████████| 9/9 [00:27<00:00,  3.11s/it]
Animation frame: 4/40  ████▏                   | 34/196 [04:07<14:50,  5.49s/it]
Creating in-between cadence frame: 2; tween:0.50;
Creating in-between cadence frame: 3; tween:1.00;
Seed: 3826518505
Prompt: tiny cute swamp bunny, highly detailed, intricate, ultra hd, sharp photo, crepuscular rays, in focus, by tomasz alen kopera
╭─────┬───┬───────┬─────┬──────┬────┬────╮
│Steps│CFG│Denoise│Angle│ Zoom │Tr X│Tr Y│
├─────┼───┼───────┼─────┼──────┼────┼────┤
│ 25  │7.0│ 0.35  │  0  │1.0035│ 0  │ 0  │
╰─────┴───┴───────┴─────┴──────┴────┴────╯
  0%|                                                     | 0/9 [00:01<?, ?it/s]
Deforum progress:  17%|████▏                   | 34/196 [04:17<20:25,  7.56s/it]
Error completing request
Arguments: ('task(srrkglpq6kkadr7)', None, False, None, '2D', 40, 'replicate', '0:(0)', '0:(1.0025+0.002*sin(1.25*3.14*t/30))', '0:(0)', '0:(0)', '0:(1.75)', '0:(0.5)', '0:(0.5)', '0:(0)', '0:(0)', '0:(0)', False, '0:(0)', '0:(0)', '0:(0)', '0:(53)', '0: (0.065)', '0: (0.65)', '0: (1.0)', '0: (7)', '0:(1.5)', False, '0:(1)', '0:(0)', False, '0: (25)', '0: (70)', '0: (1)', False, '0: (200)', '0: (10000)', '0:(s), 1:(-1), "max_f-2":(-1), "max_f-1":(s)', False, '0: ("Euler a")', '0: ("{video_mask}")', False, '0: ("{video_mask}")', False, '0: ("model1.ckpt"), 100: ("model2.safetensors")', False, '0: (2)', True, '0: (1.05)', '0: (5)', '0: (1.0)', '0: (0.1)', '0: (0.0)', 'LAB', '', 1.0, False, '2', 'None', '0: (1)', 'None', '0: (1)', '0', 'perlin', 8, 8, 4, 0.5, True, False, 0.2, 'border', 'bicubic', False, 'https://deforum.github.io/a1/V1.mp4', 1, 0, -1, False, False, 'https://deforum.github.io/a1/VM1.mp4', False, '20230129210106', '{\n    "0": "tiny cute swamp bunny, highly detailed, intricate, ultra hd, sharp photo, crepuscular rays, in focus, by tomasz alen kopera",\n    "20": "anthropomorphic clean cat, surrounded by fractals, epic angle and pose, symmetrical, 3d, depth of field, ruan jia and fenghua zhong"\n}\n    ', '', '', 1280, 768, False, False, -1, 'Euler a', False, 0, 0, 25, 0.0, 1, True, True, False, False, False, 'Deforum_{timestring}', '{timestring}_{index}_{prompt}.png', 'iter', 1, False, False, True, 0.8, 'https://deforum.github.io/a1/I1.png', False, False, False, True, 'https://deforum.github.io/a1/M1.jpg', 1.0, 1.0, 4, 1, True, 4, 'reroll', 10.0, False, 15, False, False, 'FFMPEG mp4', 'None', 'https://deforum.github.io/a1/A1.mp3', False, 'realesr-animevideov3', 'x2', True, False, 'x0_pred', 'C:/SD/20230124234916_%09d.png', 'testvidmanualsettings.mp4', False, 'None', 2, False, 2, False, None, True, False, 'None', True, 'None', False, 'RAFT', 'None', False, 'None', False, 'None', False, False, '0:(0.5)', '0:(1)', '0:(0.5)', '0:(1)', '0:(100)', '0:(0)', False, '{\n    "0": "https://deforum.github.io/a1/Gi1.png",\n    "max_f/4-5": "https://deforum.github.io/a1/Gi2.png",\n    "max_f/2-10": "https://deforum.github.io/a1/Gi3.png",\n    "3*max_f/4-15": "https://deforum.github.io/a1/Gi4.jpg",\n    "max_f-20": "https://deforum.github.io/a1/Gi1.png"\n}', '0:(0.75)', '0:(0.35)', '0:(0.25)', '0:(20)', '0:(0.075)') {}
Traceback (most recent call last):
  File "/home/stef/stable-diffusion-webui/modules/call_queue.py", line 56, in f
    res = list(func(*args, **kwargs))
  File "/home/stef/stable-diffusion-webui/modules/call_queue.py", line 37, in f
    res = func(*args, **kwargs)
  File "/home/stef/stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts/deforum.py", line 101, in run_deforum
    render_animation(args, anim_args, video_args, parseq_args, loop_args, controlnet_args, root.animation_prompts, root)
  File "/home/stef/stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts/deforum_helpers/render.py", line 573, in render_animation
    image = generate(args, keys, anim_args, loop_args, controlnet_args, root, frame_idx, sampler_name=scheduled_sampler_name)
  File "/home/stef/stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts/deforum_helpers/generate.py", line 64, in generate
    image, caught_vae_exception = generate_with_nans_check(args, keys, anim_args, loop_args, controlnet_args, root, frame, return_sample, sampler_name)
  File "/home/stef/stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts/deforum_helpers/generate.py", line 98, in generate_with_nans_check
    raise e
  File "/home/stef/stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts/deforum_helpers/generate.py", line 92, in generate_with_nans_check
    image = generate_inner(args, keys, anim_args, loop_args, controlnet_args, root, frame, return_sample, sampler_name)
  File "/home/stef/stable-diffusion-webui/extensions/deforum-for-automatic1111-webui/scripts/deforum_helpers/generate.py", line 267, in generate_inner
    processed = processing.process_images(p)
  File "/home/stef/stable-diffusion-webui/modules/processing.py", line 503, in process_images
    res = process_images_inner(p)
  File "/home/stef/stable-diffusion-webui/modules/processing.py", line 653, in process_images_inner
    samples_ddim = p.sample(conditioning=c, unconditional_conditioning=uc, seeds=seeds, subseeds=subseeds, subseed_strength=p.subseed_strength, prompts=prompts)
  File "/home/stef/stable-diffusion-webui/modules/processing.py", line 1087, in sample
    samples = self.sampler.sample_img2img(self, self.init_latent, x, conditioning, unconditional_conditioning, image_conditioning=self.image_conditioning)
  File "/home/stef/stable-diffusion-webui/modules/sd_samplers_kdiffusion.py", line 331, in sample_img2img
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "/home/stef/stable-diffusion-webui/modules/sd_samplers_kdiffusion.py", line 234, in launch_sampling
    return func()
  File "/home/stef/stable-diffusion-webui/modules/sd_samplers_kdiffusion.py", line 331, in <lambda>
    samples = self.launch_sampling(t_enc + 1, lambda: self.func(self.model_wrap_cfg, xi, extra_args=extra_args, disable=False, callback=self.callback_state, **extra_params_kwargs))
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/autograd/grad_mode.py", line 27, in decorate_context
    return func(*args, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/k-diffusion/k_diffusion/sampling.py", line 145, in sample_euler_ancestral
    denoised = model(x, sigmas[i] * s_in, **extra_args)
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/modules/sd_samplers_kdiffusion.py", line 126, in forward
    x_out = self.inner_model(x_in, sigma_in, cond=make_condition_dict([cond_in], image_cond_in))
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/k-diffusion/k_diffusion/external.py", line 112, in forward
    eps = self.get_eps(input * c_in, self.sigma_to_t(sigma), **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/k-diffusion/k_diffusion/external.py", line 138, in get_eps
    return self.inner_model.apply_model(*args, **kwargs)
  File "/home/stef/stable-diffusion-webui/modules/sd_hijack_utils.py", line 17, in <lambda>
    setattr(resolved_obj, func_path[-1], lambda *args, **kwargs: self(*args, **kwargs))
  File "/home/stef/stable-diffusion-webui/modules/sd_hijack_utils.py", line 28, in __call__
    return self.__orig_func(*args, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/models/diffusion/ddpm.py", line 858, in apply_model
    x_recon = self.model(x_noisy, t, **cond)
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1212, in _call_impl
    result = forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/models/diffusion/ddpm.py", line 1335, in forward
    out = self.diffusion_model(x, t, context=cc)
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/diffusionmodules/openaimodel.py", line 797, in forward
    h = module(h, emb, context)
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/diffusionmodules/openaimodel.py", line 84, in forward
    x = layer(x, context)
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/attention.py", line 334, in forward
    x = block(x, context=context[i])
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/attention.py", line 269, in forward
    return checkpoint(self._forward, (x, context), self.parameters(), self.checkpoint)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/diffusionmodules/util.py", line 121, in checkpoint
    return CheckpointFunction.apply(func, len(inputs), *args)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/diffusionmodules/util.py", line 136, in forward
    output_tensors = ctx.run_function(*ctx.input_tensors)
  File "/home/stef/stable-diffusion-webui/repositories/stable-diffusion-stability-ai/ldm/modules/attention.py", line 272, in _forward
    x = self.attn1(self.norm1(x), context=context if self.disable_self_attn else None) + x
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/torch/nn/modules/module.py", line 1194, in _call_impl
    return forward_call(*input, **kwargs)
  File "/home/stef/stable-diffusion-webui/modules/sd_hijack_optimizations.py", line 129, in split_cross_attention_forward
    s2 = s1.softmax(dim=-1, dtype=q.dtype)
torch.cuda.OutOfMemoryError: HIP out of memory. Tried to allocate 1.76 GiB (GPU 0; 7.98 GiB total capacity; 5.26 GiB already allocated; 2.03 GiB free; 5.89 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation.  See documentation for Memory Management and PYTORCH_HIP_ALLOC_CONF

Traceback (most recent call last):
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/blocks.py", line 987, in postprocess_data
    if predictions[i] is components._Keywords.FINISHED_ITERATING:
IndexError: tuple index out of range

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/routes.py", line 394, in run_predict
    output = await app.get_blocks().process_api(
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/blocks.py", line 1078, in process_api
    data = self.postprocess_data(fn_index, result["prediction"], state)
  File "/home/stef/stable-diffusion-webui/venv/lib/python3.10/site-packages/gradio/blocks.py", line 991, in postprocess_data
    raise ValueError(
ValueError: Number of output components does not match number of values returned from from function f

**Comment by Gcode808** [comment]
I am having the same problem, ValueError: Number of output components does not match number of values returned from from function f.... I was generating videos just fine, didnt restart, only changed the prompt. Reloaded the old prompts and double checked json format with the validator everything fine but getting this error. Have you found a fix?

**Comment by Jonglastra** [comment]
Unfortunately no, I did the same tests. i think something has been changed in deforum during the last updates (i don't think a new version of auto1111 has been released for weeks).
 What is strange is the sentence about memory:  "torch.cuda.OutOfMemoryError: HIP out of memory. Tried to allocate 1.76 GiB (GPU 0; 7.98 GiB total capacity; 5.26 GiB already allocated; 2.03 GiB free; 5.89 GiB reserved in total by PyTorch) If reserved memory is >> allocated memory try setting max_split_size_mb to avoid fragmentation) If reserved memory is >> allocated memory try setting ".
 I'm clearly under the limit of memory. I don't know how to set max_split_size_mb.
